juniper/src/types/containers.rs

```diff
@@ -257,6 +256,155 @@ where
+        let f = resolve_into_list_async(executor, info, self.iter());
+        Box::pin(f)
+    }
+}
+
+impl<T, S, const N: usize> FromInputValue<S> for [T; N]
+where
+    T: FromInputValue<S>,
+    S: ScalarValue,
+{
+    fn from_input_value(v: &InputValue<S>) -> Option<Self> {
+        match *v {
+            InputValue::List(ref ls) => {
+                // SAFETY: The `.assume_init()` here is safe, because the type
+                //         we are claiming to have initialized here is a bunch
+                //         of `MaybeUninit`s, which do not require any
+                //         initialization.
+                let mut out: [MaybeUninit<T>; N] = unsafe { MaybeUninit::uninit().assume_init() };
+                let (mut out_len, mut out_is_ok) = (0, true);
+
+                let mut items = ls.iter().filter_map(|i| i.item.convert());
+                for elem in &mut out[..] {
+                    if let Some(i) = items.next() {
+                        *elem = MaybeUninit::new(i);
+                        out_len += 1;
+                    } else {
+                        // There is not enough `items` to fill the array.
+                        out_is_ok = false;
+                        break;
+                    }
+                }
+                if items.next().is_some() {
+                    // There is too much `items` to fit into the array.
+                    out_is_ok = false;
+                }
+
+                // TODO: Use `mem::transmute` instead of `mem::transmute_copy`
+                //       below, once it's allowed for const generics:
+                //       https://github.com/rust-lang/rust/issues/61956
+                if out_is_ok {
+                    // SAFETY: `mem::transmute_copy` is safe here, because we
+                    //         have exactly `N` initialized `items`.
+                    //         Also, despite `mem::transmute_copy` copies the
+                    //         value, we won't have a double-free when `T: Drop`
+                    //         here, because original `data` is `MaybeUninit`,
+                    //         so does nothing on `Drop`.
+                    Some(unsafe { mem::transmute_copy(&out) })
+                } else {
+                    // Dropping a `MaybeUninit` does nothing, thus we need to
+                    // drop the collected `items` manually, otherwise we may
+                    // introduce a memory/resource leak if `T: Drop`.
+                    for elem in &mut out[0..out_len] {
+                        // SAFETY: This is safe, because `out_len` represents
+                        //         exactly the number of added `items`.
+                        unsafe {
+                            ptr::drop_in_place(elem.as_mut_ptr());
+                        }
+                    }
+                    None
+                }
+            }
+            ref other => {
+                other.convert().and_then(|e: T| {
```

Comment:
@LegNeato here is the input coercion in `FromInputValue` implementation. It's litterally the same as for `Vec`s, but additionally does a length check.